src/ic3/enhancers/ingress/createExtractAdaptiveCardMiddleware.ts

```diff
@@ -139,7 +165,13 @@ export default function createExtractAdaptiveCardMiddleware(): IngressMiddleware
       getState(StateKey.Logger)?.logClientSdkTelemetryEvent(Microsoft.CRM.Omnichannel.IC3Client.Model.LogLevel.WARN,
         {
           Event: TelemetryEvents.ADAPTIVE_CARD_PROCESSING_ERROR,
-          Description: `Adapter: [AdaptiveCard] Key 'attachments' not found; ignoring attachment.`
+          Description: `Adapter: [AdaptiveCard] Key 'attachments' not found; ignoring attachment.`,
+          CustomProperties: stringifyHelper({
+            activityId: activity?.id,
+            messageId: activity?.messageid,
+            [StateKey.ChatId]: getState(StateKey.ChatId),
+            [StateKey.LiveworkItemId]: getState(StateKey.LiveworkItemId)
+          })
```

Comment:
it seems like this block is repeated many times, is it possible to abstract it for reutilization?

Reply:
The parameters passing into the function are vary from one to one. If we pass in an array there is not much difference than direct calling the stringifyHelper in place. We also need to do a lot of extra type/value check. For this PR, I prefer to keep everything simple to avoid extra complicity. Once we have the telemtry in testing cluster, we can evaluate if the telemetry is sufficient. More parameters might be required.